src/components/app/ProfileSharing.js

```diff
@@ -262,20 +295,43 @@ class ProfileSharingCompositeButton extends PureComponent<
     });
   }
 
+  _onSecondSharePanelOpen() {
+    const { profileSharingStatus } = this.props;
+    this.setState({
+      shareNetworkUrls: !profileSharingStatus.sharedWithUrls,
+    });
+  }
+
   render() {
     const { state, uploadProgress, error, shortUrl } = this.state;
-    const { symbolicationStatus } = this.props;
+    const { profile, symbolicationStatus, profileSharingStatus } = this.props;
+
     const shareLabel =
       symbolicationStatus === 'DONE'
         ? 'Share...'
         : 'Sharing will be enabled once symbolication is complete';
+
+    const secondShareButtonVisibility =
+      // It must be shared before
+      state === 'public' &&
+      // If both with and without URL is shared, do not show anymore.
+      (!profileSharingStatus.sharedWithUrls ||
+        !profileSharingStatus.sharedWithoutUrls) &&
```

Comment:
Do we miss the case: loaded from public store, URLs not removed, shared without urls ?
Maybe you can set the `profileSharingStatus` state from the `VIEW_PROFILE` action from its `meta.networkURLsRemoved` value to handle this case?